### PR TITLE
[WIP] Move GA logic out of the bridge

### PR DESCRIPTION
**Pull request by samgiles**
Move logic pertaining to calling into the GA library out of the Bridge code.  The bridge code is now specific to moving data values from JavaScript to types required by the GoogleAnalytics library.

### CODE REVIEW

**Comment by samgiles** [comment]
@wilsonpage r? Once this has landed, I'll update magnet-client to use this.  I've tested that it works with these changes - and the GA class should be available outside of the React native context.

File: android/src/main/java/com/idehub/GoogleAnalyticsBridge/GoogleAnalyticsBridge.java [hunk 1]
```diff
@@ -13,371 +13,248 @@
...
+import java.util.ArrayList;
 import java.util.HashMap;
+import java.util.List;
 import java.util.Map;
 
 public class GoogleAnalyticsBridge extends ReactContextBaseJavaModule {
+        private final String _trackingId;
+        private final HashMap<String, Tracker> mTrackers
+            = new HashMap<String, Tracker>();
+        private final GA _ga;
```
Comment (wilsonpage) [inline android/src/main/java/com/idehub/GoogleAnalyticsBridge/GoogleAnalyticsBridge.java:25]: Indentation changed, is that ok?

File: android/src/main/java/com/idehub/GoogleAnalyticsBridge/GoogleAnalyticsBridge.java [hunk 2]
```diff
@@ -13,371 +13,248 @@
...
+            _ga.setOptOut(enabled);
+        }
 
-        if (analytics != null)
-        {
-            analytics.setLocalDispatchPeriod(intervalInSeconds);
+        @ReactMethod
+        public void setAppName(String trackerId, String appName){
+            _ga.setAppName(trackerId, appName);
+        }
```
Comment (wilsonpage) [inline android/src/main/java/com/idehub/GoogleAnalyticsBridge/GoogleAnalyticsBridge.java:202]: NIT: Line break between blocks.

**Review by wilsonpage — APPROVED** [review]
It's quite hard to parse the diff, but IIUC you've separated the logic from the react-native bindings. Could you outline any other major changes (if any)?

Will be interesting to see if the owner spots any issues when we PR him.

File: android/src/main/java/com/idehub/GoogleAnalyticsBridge/GoogleAnalyticsBridge.java (hunk 1 above)
Comment (samgiles) [reply android/src/main/java/com/idehub/GoogleAnalyticsBridge/GoogleAnalyticsBridge.java:25]: Was trying to make it consistent.  Should be ok.

**Comment by samgiles** [comment]
The only major change really is that I added some sensible defaults to the tracker:  https://github.com/mozilla-magnet/react-native-google-analytics-bridge/pull/1/files#diff-ab7f95c1aee91a7ea7a430b84ff6812bR40

**Comment by wilsonpage** [comment]
🚢.ly?

File: android/src/main/java/com/idehub/GoogleAnalyticsBridge/GA.java [hunk 3]
```diff
@@ -0,0 +1,334 @@
...
+    private synchronized Tracker getTracker(String trackerId) {
+        if (!_trackers.containsKey(trackerId)) {
+            GoogleAnalytics analytics = getAnalyticsInstance();
+            analytics.setLocalDispatchPeriod(LOCAL_DISPATCH_PERIOD);
+
+            Tracker tracker = analytics.newTracker(trackerId);
+            tracker.enableExceptionReporting(true);
+
+            // Disable IDFA (Identifier for Advertisers) by default
+            tracker.enableAdvertisingIdCollection(false);
```
Comment (wilsonpage) [inline android/src/main/java/com/idehub/GoogleAnalyticsBridge/GA.java]: Isn't this required to track a flow of a single user? Or is there some other unique identifier that GA uses to group events by user?
